EatGoodNaija.Server/Services/Implementation/CartRepository.cs

```diff
@@ -0,0 +1,37 @@
+﻿using EatGoodNaija.Server.Model;
+using EatGoodNaija.Server.Services.Interface;
+using StackExchange.Redis;
+using System.Text.Json;
+
+namespace EatGoodNaija.Server.Services.Implementation
+{
+    public class CartRepository : ICartRepository
+    {
+        private readonly IDatabase _database;
+        public CartRepository(IConnectionMultiplexer redis)
+        {
+            _database = redis.GetDatabase();
+        }
+        public async Task<bool> DeleteCartAsync(string cartId)
+        {
+            return await _database.KeyDeleteAsync(cartId);
+        }
+
+        public async Task<CustomerCart> GetCartAsync(string cartId)
+        {
```

Comment:
Add a service layer so that you can validate the incoming data and also handle errors with ease. Do something like this in your service;

 public async Task<CustomerCart> GetCartAsync(string cartId)
  { 
     // You can instantiate your api response here
     try
     {
        // validate cartId
      if (string.IsNullOrEmpty(cartId))
     {
        //  return your api response
     }
      var getcartData = await GetCartAsync(cartId);
     // Validate the response
      if (getcartData is null) 
        // return api response
        
     // return success response;
     }
     catch (exception ex)
     {
       // catch unexpected error that could break your code
         throw;
     }     
  }
  
  Then this service method is what you will call in your controller.
  Same goes for other methods also